File: VB/SecondaryAxesXSample/Model/SalesPerson.vb
Imports System.Data.Entity.Spatial
Imports System.ComponentModel.DataAnnotations.Schema
Imports System.ComponentModel.DataAnnotations
Imports System.Collections.Generic
Imports System

Namespace SecondaryAxesXSample

    <Table("SalesPerson")> _
    Partial Public Class SalesPerson
        <Key, Column(Order := 0), DatabaseGenerated(DatabaseGeneratedOption.None)> _
        Public Property OrderID() As Integer

        <StringLength(15)> _
        Public Property Country() As String

        <Key, Column(Order := 1), StringLength(10)> _
        Public Property FirstName() As String

        <Key, Column(Order := 2), StringLength(20)> _
        Public Property LastName() As String

        <Key, Column(Order := 3), StringLength(40)> _
        Public Property ProductName() As String

        <Key, Column(Order := 4), StringLength(15)> _
        Public Property CategoryName() As String

        Public Property OrderDate() As Date?

        <Key, Column(Order := 5, TypeName := "smallmoney")> _
        Public Property UnitPrice() As Decimal

        <Key, Column(Order := 6), DatabaseGenerated(DatabaseGeneratedOption.None)> _
        Public Property Quantity() As Short

        <Key, Column(Order := 7)> _
        Public Property Discount() As Single

        <Column("Extended Price", TypeName := "money")> _
        Public Property ExtendedPrice() As Decimal?

        <Key, Column("Sales Person", Order := 8), StringLength(31)> _
        Public Property Sales_Person() As String
    End Class
End Namespace

File: CS/SecondaryAxesXSample/Model/SalesPerson.cs
namespace SecondaryAxesXSample
{
    using System;
    using System.Collections.Generic;
    using System.ComponentModel.DataAnnotations;
    using System.ComponentModel.DataAnnotations.Schema;
    using System.Data.Entity.Spatial;

    [Table("SalesPerson")]
    public partial class SalesPerson
    {
        [Key]
        [Column(Order = 0)]
        [DatabaseGenerated(DatabaseGeneratedOption.None)]
        public int OrderID { get; set; }

        [StringLength(15)]
        public string Country { get; set; }

        [Key]
        [Column(Order = 1)]
        [StringLength(10)]
        public string FirstName { get; set; }

        [Key]
        [Column(Order = 2)]
        [StringLength(20)]
        public string LastName { get; set; }

        [Key]
        [Column(Order = 3)]
        [StringLength(40)]
        public string ProductName { get; set; }

        [Key]
        [Column(Order = 4)]
        [StringLength(15)]
        public string CategoryName { get; set; }

        public DateTime? OrderDate { get; set; }

        [Key]
        [Column(Order = 5, TypeName = "smallmoney")]
        public decimal UnitPrice { get; set; }

        [Key]
        [Column(Order = 6)]
        [DatabaseGenerated(DatabaseGeneratedOption.None)]
        public short Quantity { get; set; }

        [Key]
        [Column(Order = 7)]
        public float Discount { get; set; }

        [Column("Extended Price", TypeName = "money")]
        public decimal? ExtendedPrice { get; set; }

        [Key]
        [Column("Sales Person", Order = 8)]
        [StringLength(31)]
        public string Sales_Person { get; set; }
    }
}
